### ISSUE TITLE
npm install with git dependencies fails on windows

### ISSUE THREAD

**Issue by h2oboi89** [npm/npm#10729]
package.json:

```
{
   "name": "gea-communication",
   "version": "0.0.0",
   "description": "GEA2 Communication",
   "main": "index.js",
   "scripts": {
      "gulp": "gulp",
      "lint": "gulp lint",
      "test": "gulp test",
      "coverage": "gulp coverage",
      "watch": "gulp watch"
   },
   "repository": {
      "type": "git",
      "url": "git@github.appl.ge.com:PCToolsJavascript/geaCommunication.git"
   },
   "author": "",
   "license": "ISC",
   "devDependencies": {
      "gulp": "^3.9.0",
      "gulp-istanbul": "^0.10.2",
      "gulp-jasmine": "^2.2.0",
      "gulp-jshint": "^1.12.0",
      "gulp-plumber": "^1.0.1",
      "gulp-tap": "^0.1.3",
      "gulp-watch": "^4.3.5",
      "mach.js": "0.0.8",
      "doubles": "git+https://github.appl.ge.com/PCToolsJavascript/doubles.git",
      "buffertools": "^2.1.3"
   },
   "dependencies": {
      "javascript-common": "git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git",
      "usb-adapter": "git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git",
      "byte-stream": "git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git"
   }
}
```

npm debug log:

```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install' ]
2 info using npm@3.3.6
3 info using node@v5.0.0
4 silly loadCurrentTree Starting
5 silly install loadCurrentTree
6 silly install readLocalPackageData
7 silly install normalizeTree
8 silly loadCurrentTree Finishing
9 silly loadIdealTree Starting
10 silly install loadIdealTree
11 silly cloneCurrentTree Starting
12 silly install cloneCurrentTreeToIdealTree
13 silly cloneCurrentTree Finishing
14 silly loadShrinkwrap Starting
15 silly install loadShrinkwrap
16 silly loadShrinkwrap Finishing
17 silly loadAllDepsIntoIdealTree Starting
18 silly install loadAllDepsIntoIdealTree
19 silly fetchOtherPackageData javascript-common@git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git
20 silly cache add args [ 'javascript-common@git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git',
20 silly cache add   null ]
21 verbose cache add spec javascript-common@git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git
22 silly fetchOtherPackageData usb-adapter@git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git
23 silly cache add args [ 'usb-adapter@git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git',
23 silly cache add   null ]
24 verbose cache add spec usb-adapter@git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git
25 silly fetchOtherPackageData byte-stream@git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git
26 silly cache add args [ 'byte-stream@git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git',
26 silly cache add   null ]
27 verbose cache add spec byte-stream@git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git
28 silly cache add parsed spec Result {
28 silly cache add   raw: 'javascript-common@git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git',
28 silly cache add   scope: null,
28 silly cache add   name: 'javascript-common',
28 silly cache add   rawSpec: 'git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git',
28 silly cache add   spec: 'https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git',
28 silly cache add   type: 'git' }
29 verbose addRemoteGit caching git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git
30 silly tryClone cloning git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git via git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git
31 verbose tryClone git-https-github-appl-ge-com-PCToolsJavascript-javascriptCommon-git-be4c00f873049c36ca16b30710a81b02 not in flight; caching
32 silly cache add parsed spec Result {
32 silly cache add   raw: 'usb-adapter@git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git',
32 silly cache add   scope: null,
32 silly cache add   name: 'usb-adapter',
32 silly cache add   rawSpec: 'git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git',
32 silly cache add   spec: 'https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git',
32 silly cache add   type: 'git' }
33 verbose addRemoteGit caching git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git
34 silly tryClone cloning git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git via git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git
35 verbose tryClone git-https-github-appl-ge-com-PCToolsJavascript-usbAdapter-git-a6c1b0df83aa211d23bb5a293cdec37e not in flight; caching
36 silly cache add parsed spec Result {
36 silly cache add   raw: 'byte-stream@git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git',
36 silly cache add   scope: null,
36 silly cache add   name: 'byte-stream',
36 silly cache add   rawSpec: 'git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git',
36 silly cache add   spec: 'https://github.appl.ge.com/PCToolsJavascript/byte-stream.git',
36 silly cache add   type: 'git' }
37 verbose addRemoteGit caching git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git
38 silly tryClone cloning git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git via git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git
39 verbose tryClone git-https-github-appl-ge-com-PCToolsJavascript-byte-stream-git-5bf1e8790d37f9eef914fae2780aaa6a not in flight; caching
40 verbose makeDirectory C:\Users\220040509\AppData\Roaming\npm-cache\_git-remotes creation not in flight; initializing
41 verbose makeDirectory C:\Users\220040509\AppData\Roaming\npm-cache\_git-remotes creation already in flight; waiting
42 verbose makeDirectory C:\Users\220040509\AppData\Roaming\npm-cache\_git-remotes creation already in flight; waiting
43 verbose makeCacheDir UID & GID are irrelevant on win32
44 info git [ 'clone',
44 info git   '--template=C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\_templates',
44 info git   '--mirror',
44 info git   'https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git',
44 info git   'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\git-https-github-appl-ge-com-PCToolsJavascript-usbAdapter-git-a6c1b0df83aa211d23bb5a293cdec37e' ]
45 info git [ 'clone',
45 info git   '--template=C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\_templates',
45 info git   '--mirror',
45 info git   'https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git',
45 info git   'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\git-https-github-appl-ge-com-PCToolsJavascript-javascriptCommon-git-be4c00f873049c36ca16b30710a81b02' ]
46 info git [ 'clone',
46 info git   '--template=C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\_templates',
46 info git   '--mirror',
46 info git   'https://github.appl.ge.com/PCToolsJavascript/byte-stream.git',
46 info git   'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\git-https-github-appl-ge-com-PCToolsJavascript-byte-stream-git-5bf1e8790d37f9eef914fae2780aaa6a' ]
47 verbose mirrorRemote git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git git clone https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git
48 verbose setPermissions git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git skipping chownr on Windows
49 verbose resolveHead git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git original treeish: master
50 info git [ 'rev-list', '-n1', 'master' ]
51 verbose mirrorRemote git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git git clone https://github.appl.ge.com/PCToolsJavascript/byte-stream.git
52 verbose setPermissions git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git skipping chownr on Windows
53 verbose resolveHead git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git original treeish: master
54 info git [ 'rev-list', '-n1', 'master' ]
55 silly resolveHead git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git resolved treeish: 9f14ddc283dbe11e1c1eacf70052de1ad636c1ec
56 verbose resolveHead git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git resolved Git URL: git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git#9f14ddc283dbe11e1c1eacf70052de1ad636c1ec
57 silly resolveHead Git working directory: C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-dfee2d4f230f2cc61b3dd6e11157a972\9f14ddc283dbe11e1c1eacf70052de1ad636c1ec
58 silly resolveHead git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git resolved treeish: 089b6c6758e60316fec186a0e3c260fb02bd2ef7
59 verbose resolveHead git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git resolved Git URL: git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git#089b6c6758e60316fec186a0e3c260fb02bd2ef7
60 silly resolveHead Git working directory: C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-31220edb0c378fcd37e9e32433e19120\089b6c6758e60316fec186a0e3c260fb02bd2ef7
61 info git [ 'clone',
61 info git   'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\git-https-github-appl-ge-com-PCToolsJavascript-usbAdapter-git-a6c1b0df83aa211d23bb5a293cdec37e',
61 info git   'C:\\Users\\220040~1\\AppData\\Local\\Temp\\npm-11656-245c85b4\\git-cache-dfee2d4f230f2cc61b3dd6e11157a972\\9f14ddc283dbe11e1c1eacf70052de1ad636c1ec' ]
62 info git [ 'clone',
62 info git   'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\git-https-github-appl-ge-com-PCToolsJavascript-byte-stream-git-5bf1e8790d37f9eef914fae2780aaa6a',
62 info git   'C:\\Users\\220040~1\\AppData\\Local\\Temp\\npm-11656-245c85b4\\git-cache-31220edb0c378fcd37e9e32433e19120\\089b6c6758e60316fec186a0e3c260fb02bd2ef7' ]
63 verbose mirrorRemote git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git git clone https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git
64 verbose setPermissions git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git skipping chownr on Windows
65 verbose resolveHead git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git original treeish: master
66 info git [ 'rev-list', '-n1', 'master' ]
67 silly resolveHead git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git resolved treeish: a912a2ea2e3a77af99591eb54ede523aca893459
68 verbose resolveHead git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git resolved Git URL: git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git#a912a2ea2e3a77af99591eb54ede523aca893459
69 silly resolveHead Git working directory: C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-05a1b2840b452b39e46674cf49bed9fd\a912a2ea2e3a77af99591eb54ede523aca893459
70 info git [ 'clone',
70 info git   'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\_git-remotes\\git-https-github-appl-ge-com-PCToolsJavascript-javascriptCommon-git-be4c00f873049c36ca16b30710a81b02',
70 info git   'C:\\Users\\220040~1\\AppData\\Local\\Temp\\npm-11656-245c85b4\\git-cache-05a1b2840b452b39e46674cf49bed9fd\\a912a2ea2e3a77af99591eb54ede523aca893459' ]
71 verbose cloneResolved git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git clone Cloning into 'C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-dfee2d4f230f2cc61b3dd6e11157a972\9f14ddc283dbe11e1c1eacf70052de1ad636c1ec'...
71 verbose cloneResolved done.
72 verbose cloneResolved git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git clone Cloning into 'C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-31220edb0c378fcd37e9e32433e19120\089b6c6758e60316fec186a0e3c260fb02bd2ef7'...
72 verbose cloneResolved done.
73 info git [ 'checkout', '9f14ddc283dbe11e1c1eacf70052de1ad636c1ec' ]
74 info git [ 'checkout', '089b6c6758e60316fec186a0e3c260fb02bd2ef7' ]
75 verbose checkoutTreeish git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git checkout Note: checking out '089b6c6758e60316fec186a0e3c260fb02bd2ef7'.
75 verbose checkoutTreeish
75 verbose checkoutTreeish You are in 'detached HEAD' state. You can look around, make experimental
75 verbose checkoutTreeish changes and commit them, and you can discard any commits you make in this
75 verbose checkoutTreeish state without impacting any branches by performing another checkout.
75 verbose checkoutTreeish
75 verbose checkoutTreeish If you want to create a new branch to retain commits you create, you may
75 verbose checkoutTreeish do so (now or later) by using -b with the checkout command again. Example:
75 verbose checkoutTreeish
75 verbose checkoutTreeish   git checkout -b <new-branch-name>
75 verbose checkoutTreeish
75 verbose checkoutTreeish HEAD is now at 089b6c6... Merge branch 'master' of github.appl.ge.com:PCToolsJavascript/byte-stream
76 verbose checkoutTreeish git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git checkout Note: checking out '9f14ddc283dbe11e1c1eacf70052de1ad636c1ec'.
76 verbose checkoutTreeish
76 verbose checkoutTreeish You are in 'detached HEAD' state. You can look around, make experimental
76 verbose checkoutTreeish changes and commit them, and you can discard any commits you make in this
76 verbose checkoutTreeish state without impacting any branches by performing another checkout.
76 verbose checkoutTreeish
76 verbose checkoutTreeish If you want to create a new branch to retain commits you create, you may
76 verbose checkoutTreeish do so (now or later) by using -b with the checkout command again. Example:
76 verbose checkoutTreeish
76 verbose checkoutTreeish   git checkout -b <new-branch-name>
76 verbose checkoutTreeish
76 verbose checkoutTreeish HEAD is now at 9f14ddc... Merge pull request #39 from PCToolsJavascript/ConnectOnlyAfterBind
77 verbose addLocalDirectory C:\Users\220040509\AppData\Roaming\npm-cache\byte-stream\0.0.0\package.tgz not in flight; packing
78 verbose addLocalDirectory C:\Users\220040509\AppData\Roaming\npm-cache\usb-adapter\0.0.1\package.tgz not in flight; packing
79 verbose tar pack [ 'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\byte-stream\\0.0.0\\package.tgz',
79 verbose tar pack   'C:\\Users\\220040~1\\AppData\\Local\\Temp\\npm-11656-245c85b4\\git-cache-31220edb0c378fcd37e9e32433e19120\\089b6c6758e60316fec186a0e3c260fb02bd2ef7' ]
80 verbose tarball C:\Users\220040509\AppData\Roaming\npm-cache\byte-stream\0.0.0\package.tgz
81 verbose folder C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-31220edb0c378fcd37e9e32433e19120\089b6c6758e60316fec186a0e3c260fb02bd2ef7
82 verbose tar pack [ 'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\usb-adapter\\0.0.1\\package.tgz',
82 verbose tar pack   'C:\\Users\\220040~1\\AppData\\Local\\Temp\\npm-11656-245c85b4\\git-cache-dfee2d4f230f2cc61b3dd6e11157a972\\9f14ddc283dbe11e1c1eacf70052de1ad636c1ec' ]
83 verbose tarball C:\Users\220040509\AppData\Roaming\npm-cache\usb-adapter\0.0.1\package.tgz
84 verbose folder C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-dfee2d4f230f2cc61b3dd6e11157a972\9f14ddc283dbe11e1c1eacf70052de1ad636c1ec
85 error addLocal Could not install C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-31220edb0c378fcd37e9e32433e19120\089b6c6758e60316fec186a0e3c260fb02bd2ef7
86 silly fetchPackageMetaData Error: ENOENT: no such file or directory, scandir 'C:\Users\220040509\AppData\Roaming\npm-cache\byte-stream\0.0.0\package.tgz'
86 silly fetchPackageMetaData     at Error (native)
86 silly fetchPackageMetaData  error for byte-stream@git+https://github.appl.ge.com/PCToolsJavascript/byte-stream.git { [Error: ENOENT: no such file or directory, scandir 'C:\Users\220040509\AppData\Roaming\npm-cache\byte-stream\0.0.0\package.tgz']
86 silly fetchPackageMetaData   errno: -4058,
86 silly fetchPackageMetaData   code: 'ENOENT',
86 silly fetchPackageMetaData   syscall: 'scandir',
86 silly fetchPackageMetaData   path: 'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\byte-stream\\0.0.0\\package.tgz' }
87 error addLocal Could not install C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-dfee2d4f230f2cc61b3dd6e11157a972\9f14ddc283dbe11e1c1eacf70052de1ad636c1ec
88 silly fetchPackageMetaData Error: ENOENT: no such file or directory, scandir 'C:\Users\220040509\AppData\Roaming\npm-cache\usb-adapter\0.0.1\package.tgz'
88 silly fetchPackageMetaData     at Error (native)
88 silly fetchPackageMetaData  error for usb-adapter@git+https://github.appl.ge.com/PCToolsJavascript/usbAdapter.git { [Error: ENOENT: no such file or directory, scandir 'C:\Users\220040509\AppData\Roaming\npm-cache\usb-adapter\0.0.1\package.tgz']
88 silly fetchPackageMetaData   errno: -4058,
88 silly fetchPackageMetaData   code: 'ENOENT',
88 silly fetchPackageMetaData   syscall: 'scandir',
88 silly fetchPackageMetaData   path: 'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\usb-adapter\\0.0.1\\package.tgz' }
89 verbose cloneResolved git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git clone Cloning into 'C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-05a1b2840b452b39e46674cf49bed9fd\a912a2ea2e3a77af99591eb54ede523aca893459'...
89 verbose cloneResolved done.
90 info git [ 'checkout', 'a912a2ea2e3a77af99591eb54ede523aca893459' ]
91 verbose checkoutTreeish git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git checkout Note: checking out 'a912a2ea2e3a77af99591eb54ede523aca893459'.
91 verbose checkoutTreeish
91 verbose checkoutTreeish You are in 'detached HEAD' state. You can look around, make experimental
91 verbose checkoutTreeish changes and commit them, and you can discard any commits you make in this
91 verbose checkoutTreeish state without impacting any branches by performing another checkout.
91 verbose checkoutTreeish
91 verbose checkoutTreeish If you want to create a new branch to retain commits you create, you may
91 verbose checkoutTreeish do so (now or later) by using -b with the checkout command again. Example:
91 verbose checkoutTreeish
91 verbose checkoutTreeish   git checkout -b <new-branch-name>
91 verbose checkoutTreeish
91 verbose checkoutTreeish HEAD is now at a912a2e... Minor edits
92 verbose addLocalDirectory C:\Users\220040509\AppData\Roaming\npm-cache\javascript-common\0.0.0\package.tgz not in flight; packing
93 verbose tar pack [ 'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\javascript-common\\0.0.0\\package.tgz',
93 verbose tar pack   'C:\\Users\\220040~1\\AppData\\Local\\Temp\\npm-11656-245c85b4\\git-cache-05a1b2840b452b39e46674cf49bed9fd\\a912a2ea2e3a77af99591eb54ede523aca893459' ]
94 verbose tarball C:\Users\220040509\AppData\Roaming\npm-cache\javascript-common\0.0.0\package.tgz
95 verbose folder C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-05a1b2840b452b39e46674cf49bed9fd\a912a2ea2e3a77af99591eb54ede523aca893459
96 error addLocal Could not install C:\Users\220040~1\AppData\Local\Temp\npm-11656-245c85b4\git-cache-05a1b2840b452b39e46674cf49bed9fd\a912a2ea2e3a77af99591eb54ede523aca893459
97 silly fetchPackageMetaData Error: ENOENT: no such file or directory, scandir 'C:\Users\220040509\AppData\Roaming\npm-cache\javascript-common\0.0.0\package.tgz'
97 silly fetchPackageMetaData     at Error (native)
97 silly fetchPackageMetaData  error for javascript-common@git+https://github.appl.ge.com/PCToolsJavascript/javascriptCommon.git { [Error: ENOENT: no such file or directory, scandir 'C:\Users\220040509\AppData\Roaming\npm-cache\javascript-common\0.0.0\package.tgz']
97 silly fetchPackageMetaData   errno: -4058,
97 silly fetchPackageMetaData   code: 'ENOENT',
97 silly fetchPackageMetaData   syscall: 'scandir',
97 silly fetchPackageMetaData   path: 'C:\\Users\\220040509\\AppData\\Roaming\\npm-cache\\javascript-common\\0.0.0\\package.tgz' }
98 silly rollbackFailedOptional Starting
99 silly rollbackFailedOptional Finishing
100 silly runTopLevelLifecycles Starting
101 silly runTopLevelLifecycles Finishing
102 silly install printInstalled
103 verbose stack Error: ENOENT: no such file or directory, scandir 'C:\Users\220040509\AppData\Roaming\npm-cache\byte-stream\0.0.0\package.tgz'
103 verbose stack     at Error (native)
104 verbose cwd C:\Users\220040509\workspace\ApplianceTools\JavaScript\gea-communication
105 error Windows_NT 6.1.7601
106 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
107 error node v5.0.0
108 error npm  v3.3.6
109 error path C:\Users\220040509\AppData\Roaming\npm-cache\byte-stream\0.0.0\package.tgz
110 error code ENOENT
111 error errno -4058
112 error syscall scandir
113 error enoent ENOENT: no such file or directory, scandir 'C:\Users\220040509\AppData\Roaming\npm-cache\byte-stream\0.0.0\package.tgz'
113 error enoent This is most likely not a problem with npm itself
113 error enoent and is related to npm not being able to find a file.
114 verbose exit [ -4058, true ]
```

**Comment by h2oboi89** [comment]
Browsing to location in file browser clearly reveals that file is present
![capture](https://cloud.githubusercontent.com/assets/5531999/11630833/44469cf0-9ccd-11e5-96a4-78807f975e46.PNG)

**Comment by parabolox** [comment]
I was getting a similar error when attempting to install Osprey, specifically when it tried to install its router dependency. On Windows 7 with npm@2.14.12 and node@v4.2.6, I would run the following command for example and get the same error (ENOENT when doing a scandir even though the file exists):

<pre>npm cache add router@git+https://github.com/blakeembrey/router.git#router-engine -ddd</pre>

After adding a bunch of logging messages, it seemed to fail around some chown/chmod/stat commands. For practical purposes I implemented the following patch to the NPM libraries locally which forks around them if it's running on Windows:

In "C:\Program Files\nodejs\node_modules\npm\lib\cache\add-local.js" in ***addLocalDirectory*** function definition _replace_:
<code>chownr(made || tgz, cs.uid, cs.gid, wrapped)</code>
_with_:
<pre><code>// Skip over mode change in Windows; we don't even care about read-only
if(process.platform === 'win32') {wrapped(); return undefined;} chownr(made || tgz, cs.uid, cs.gid, wrapped)</code></pre>

Similarly in "C:\Program Files\nodejs\node_modules\npm\lib\cache\add-remote-git.js" in ***addModeRecursive*** function definition, _add as first lines of function code_:
<pre><code>// Skip over mode change in Windows; we don't even care about read-only
if(process.platform === 'win32') {cb(); return undefined;}</code></pre>

**Comment by parabolox** [comment]
For what it's worth, the error did not seem to occur on my Windows 10 machine with the same Node and npm versions. I have a hunch that the Digital Guardian/Verdasys file system driver (which is installed on my developer Windows 7 machine but not on my Windows 10 machine) might be doing something "extra" that interferes with the operation of low-level scandir function (_fs__scandir_ in node/deps/uv/src/win/fs.c), but I haven't yet been able to verify that. I was able to rule out McAfee On-Access scanning, though.

**Comment by parabolox** [comment]
Alternatively, you can patch libuv's fs__scandir function; add this code immediately before the dir_handle is created with CreateFileW:<pre><code>  // Check if it is a directory early so that some legacy file system filter
  // drivers (like Digital Guardian/Verdasys loss prevention software)
  // don't return "file not found" errors on files that exist
  // but are not directories
  unsigned long directoryCheck = GetFileAttributesW(req->file.pathw);
  if (directoryCheck == INVALID_FILE_ATTRIBUTES) {
    goto win32_error;
  }
  else if (!(directoryCheck & FILE_ATTRIBUTE_DIRECTORY)) {
	goto not_a_directory_error;
  }</code></pre>

**Comment by geoarchitect** [comment]
Experienced the same issue with a similar setup. I was working on this for a while before I stumbled on this issue. I also have a similar setup on one of my machines with Digital Guardian and windows 8.1

Making those two changes fixed it for me.

Also Note: 

I had to make these changes in my %appdata%/roaming/npm/node_modules/npm/lib/cache directory accordingly as that's where my latest version of npm lives.

**Comment by parabolox** [comment]
Thanks for the confirmation, @geoarchitect ! I also got confirmation from @h2oboi89 that the fix worked as well.

Separately I created a proof-of-concept test program (unfortunately proprietary) that demonstrates that the NtQueryDirectoryFile API is returning an improper value when provided a file path (instead of directory path) with Digital Guardian enabled. My company's team will be working the issue and escalate it to DG if appropriate. I'll try to post back here once some resolution is achieved.

**Comment by zkat** [comment]
This seems like it might be related to #12389 somehow.

**Comment by orendin** [comment]
Hi all

I have experienced the same issue on our virtual build server and I suspect it is connected with some kind of company protection policy similar to your Digital Guardian. The dealings I had with IT so far on other matters left me with little hope of getting this resolved via IT department, though.
Your scripting solution works (thanks to @geoarchitect, I had to use the path you provided) and it works. However, I am not a friend of this approach, since the next time npm is upgraded on the build server is rebuilt, the script changes will be gone. Developer notes might help, but it is still a potential source of error.
The temporary solution I have chosen, which might or might not be relevant for you, is to include our referenced library as a _git submodule_, since we don't have problems using (private) git in general, only with npm into git.
From there, we could locally reference the library, which worked ok.

Maybe this option might help someone else, too.
